### PR TITLE
[Optimize](simd json reader) Cached search results for previous row (keyed as index in JSON object) - used as a hint.

### PR DESCRIPTION
**Pull request by eldenmoon**
# Proposed changes

optimize load cost: 78s->62s

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by eldenmoon** [comment]
run buildall

File: be/src/vec/exec/format/json/new_json_reader.cpp [hunk 1]
```diff
@@ -1042,9 +1042,10 @@ Status NewJsonReader::_simdjson_init_reader() {
     }
     _ondemand_json_parser = std::make_unique<simdjson::ondemand::parser>();
     for (int i = 0; i < _file_slot_descs.size(); ++i) {
-        _slot_desc_index.emplace(_file_slot_descs[i]->col_name(), i);
+        _slot_desc_index[_file_slot_descs[i]->col_name()] = i;
```
Comment (xiaokang) [inline be/src/vec/exec/format/json/new_json_reader.cpp:1045]: what's the difference?

**Review by xiaokang — APPROVED** [review]
LGTM

File: be/src/vec/exec/format/json/new_json_reader.cpp (hunk 1 above)
Comment (eldenmoon) [reply be/src/vec/exec/format/json/new_json_reader.cpp:1045]: no difference

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**